src/visitor.rs

```diff
@@ -683,7 +683,8 @@ impl<'b, 'a: 'b> FmtVisitor<'a> {
 
         // 1 = ;
         let shape = self.shape().saturating_sub_width(1);
```

Comment:
I'd like to update `visit_mac` as follows:

```rust
    fn visit_mac(&mut self, mac: &ast::MacCall, ident: Option<symbol::Ident>, pos: MacroPosition) {
        skip_out_of_file_lines_range_visitor!(self, mac.span());

        // 1 = ;
        let shape = self.shape().saturating_sub_width(1);
        let original_style = macro_style(mac, &self.get_context());
        let (rewrite, rewrite_style) =
            self.with_context(|ctx| rewrite_macro(mac, ident, ctx, shape, pos));
        // As of v638 of the rustc-ap-* crates, the associated span no longer includes
        // the trailing semicolon. This determines the correct span to ensure scenarios
        // with whitespace between the delimiters and trailing semi (i.e. `foo!(abc)     ;`)
        // are formatted correctly.
        let rewrite = match rewrite_style.unwrap_or(original_style) {
            Delimiter::Bracket | Delimiter::Parenthesis if MacroPosition::Item == pos => rewrite
                .map(|rw| {
                    if !rw.ends_with(';') {
                        format!("{};", rw)
                    } else {
                        rw
                    }
                }),
            _ => rewrite,
        };

        let span = match original_style {
            Delimiter::Bracket | Delimiter::Parenthesis if MacroPosition::Item == pos => {
                let search_span = mk_sp(mac.span().hi(), self.snippet_provider.end_pos());
                let hi = self.snippet_provider.span_before(search_span, ";");
                mk_sp(mac.span().lo(), hi + BytePos(1))
            }
            _ => mac.span(),
        };

        self.push_rewrite(span, rewrite);
    }
```

First, we match on the the new delimiters we got back from the rewrite, falling back to the original macro delimiters if necessary. On our second match we want to match on the original delimiters so we can properly update the span. Your approach did this, but it duplicated some of the logic from the first match.

Reply:
Need some additional tests